MapboxNavigation/CarPlayManager.swift

```diff
@@ -479,7 +479,7 @@ extension CarPlayManager: CPMapTemplateDelegate {
         let route = routeChoice.userInfo as! Route
         var service: NavigationService
         
-        if let override = delegate?.carPlayManager?(self, navigationServiceAlong: route) {
+        if let override = delegate?.carPlayManager(self, navigationServiceAlong: route) {
             service = override
         } else {
             service = MapboxNavigationService(route: route, simulating: simulatesLocations ? .always : .onPoorGPS)
```

Comment:
Ultimately, we will be moving away from ad hoc creation of a service to something else, right? Retrieval from a registry, forced dependency injection, magic pixie dust... ?